.github/workflows/main.yml

```diff
@@ -88,14 +88,22 @@ jobs:
       if: runner.os == 'Windows'
       env:
         DUB: ${{ github.workspace }}\bin\dub.exe
-      # Only run `dub test` to run unittests so far,
-      # the test-suite needs to be overhauled to support Windows
       run: |
         dub build --compiler=${{ env.DC }}
         if [[ ${{ matrix.do_test }} == 'true' ]]; then
           dub test  --compiler=${{ env.DC }}
           dub run   --compiler=${{ env.DC }} --single test/issue2051_running_unittests_from_dub_single_file_packages_fails.d
-          cd test
-          dub --single run-unittest.d
+          dub --single test/run-unittest.d
+
+          # FIXME: DMD fails a few tests on Windows; remove them for now
+          if [[ '${{ matrix.dc }}' = dmd* ]]; then
+            # DLL support is lacking
+            rm -rf test/{1-dynLib-simple,2-dynLib-dep,2-dynLib-with-staticLib-dep}
+            # Unicode in paths too
+            rm -rf test/issue130-unicode-СНА*
+            # ImportC probably requires set-up MSVC environment variables
+            rm -rf test/use-c-sources
+          fi
+          test/run-unittest.sh
```

Comment:
I have worked on this subject in the past. Check out my outcomes below:
https://github.com/dlang/dub/compare/master...shoo:dub:enh_test_windows

The difference from this PR's approach is that in my approach I changed `run-unittest.d` to be more like `run-unittest.sh` and not dependent on `.sh` .

In windows, there are several ways to run `.sh` . Cygwin(or MSYS2) / git-bash / WSL
GitHub Actions uses git-bash, which I think should be avoided for compatibility reasons.

Reply:
AFAIK, the gnu tools bundled with git for windows originate from msys2. Why do you think it should be avoided for compatibility reasons? Since almost any Windows dev has git, he/she has bash + gnu tools too. I've been using bash on Windows for years, and it's absolutely okay, especially for such trivial stuff that is easier handled in bash than D.